Configuration not fully validated before starting the experiment

Currently, when we start the Everest server we also start the optimization process, and the path of the main config file is passed to the server. This means that parts of the config is only validated on the server, we should instead pass the complete, validated, configuration to the server to start the experiment.

Another side effect of this is that an empty configuration file is considered valid in Everest, but fails once submitted to the server, take this test:

```python
@pytest.mark.timeout(60)  # Simulation might not finish
@pytest.mark.integration_test
@pytest.mark.xdist_group(name="starts_everest")
async def test_starting_not_in_folder(tmp_path, monkeypatch):
    async def server_running():
        while True:
            event = await driver.event_queue.get()
            if isinstance(event, FinishedEvent) and event.iens == 0:
                return event

    os.makedirs(tmp_path / "new_folder")
    monkeypatch.chdir(tmp_path / "new_folder")
    everest_config = EverestConfig.with_defaults()
    everest_config.dump("minimal_config.yml")
    everest_config.config_path = Path("minimal_config.yml").absolute()
    makedirs_if_needed(everest_config.output_dir, roll_if_exists=True)
    driver = await start_server(everest_config, debug=True)
    final_state = await server_running()
    assert final_state.returncode == 0
    status = everserver_status(everest_config)
    assert status["status"] == ServerStatus.completed
```

which failed on the server with:
```
Optimization run failed, with error: Traceback (most recent call last):
  File "/data/projects/ert/src/everest/detached/jobs/everserver.py", line 278, in main
    run_model = EverestRunModel.create(
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/projects/ert/src/ert/run_models/everest_run_model.py", line 381, in create
    ert_config = everest_to_ert_config(cls._add_defaults(ever_config))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/projects/ert/src/everest/simulator/everest_to_ert.py", line 510, in everest_to_ert_config
    config_dict = _everest_to_ert_config_dict(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/projects/ert/src/everest/simulator/everest_to_ert.py", line 496, in _everest_to_ert_config_dict
    _extract_forward_model(ever_config, ert_config)
  File "/data/projects/ert/src/everest/simulator/everest_to_ert.py", line 434, in _extract_forward_model
    forward_model = _extract_data_operations(ever_config)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/projects/ert/src/everest/simulator/everest_to_ert.py", line 371, in _extract_data_operations
    is_dir = _is_dir_all_geo(source, ever_config)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/projects/ert/src/everest/simulator/everest_to_ert.py", line 329, in _is_dir_all_geo
    raise AssertionError(msg)
AssertionError: Expected realizations when analysing data installation source
```

I would expect this to be validated up front, and never dispatched to the server.

--- a/src/everest/bin/everest_script.py
+++ b/src/everest/bin/everest_script.py
@@ -18,6 +18,7 @@ from everest.detached import (
     start_server,
     wait_for_server,
 )
+from everest.simulator.everest_to_ert import everest_to_ert_config
 from everest.strings import EVEREST
 from everest.util import (
     makedirs_if_needed,
@@ -50,6 +51,12 @@ def everest_entry(args=None):
             signal.SIGINT,
             partial(handle_keyboard_interrupt, options=options),
         )
+
+    # Validate ert config
+    try:
+        _ = everest_to_ert_config(options.config)
+    except ValueError as exc:
+        raise SystemExit(f"Config validation error: {exc}") from exc
 
     if EverestRunModel.create(options.config).check_if_runpath_exists():
         warn_user_that_runpath_is_nonempty()

--- a/src/everest/simulator/everest_to_ert.py
+++ b/src/everest/simulator/everest_to_ert.py
@@ -326,7 +326,7 @@ def _is_dir_all_geo(source, ever_config: EverestConfig):
     realizations = ever_config.model.realizations
     if not realizations:
         msg = "Expected realizations when analysing data installation source"
-        raise AssertionError(msg)
+        raise ValueError(msg)
 
     is_dir = []
     for geo_id in realizations:
@@ -336,7 +336,7 @@ def _is_dir_all_geo(source, ever_config: EverestConfig):
                 "Expected source to exist for data installation, "
                 "did not find: {}".format(geo_source)
             )
-            raise AssertionError(msg)
+            raise ValueError(msg)
 
         is_dir.append(os.path.isdir(geo_source))
 

--- a/tests/everest/entry_points/test_everest_entry.py
+++ b/tests/everest/entry_points/test_everest_entry.py
@@ -1,6 +1,7 @@
 import logging
 import os
 from functools import partial
+from pathlib import Path
 from unittest.mock import patch
 
 import pytest
@@ -496,3 +497,21 @@ def test_complete_status_for_normal_run_monitor(
 
     assert expected_status == status["status"]
     assert expected_error == status["message"]
+
+
+@patch("everest.bin.everest_script.server_is_running", return_value=False)
+@patch(
+    "everest.bin.everest_script.everserver_status",
+    return_value={"status": ServerStatus.never_run, "message": None},
+)
+async def test_validate_ert_config_before_starting_everest_server(
+    tmp_path, monkeypatch
+):
+    os.makedirs(tmp_path / "new_folder")
+    monkeypatch.chdir(tmp_path / "new_folder")
+    everest_config = EverestConfig.with_defaults()
+    everest_config.dump("minimal_config.yml")
+    everest_config.config_path = Path("minimal_config.yml").absolute()
+    error = "Expected realizations when analysing data installation source"
+    with pytest.raises(SystemExit, match=f"Config validation error: {error}"):
+        everest_entry([str(everest_config.config_path)])

